jaeger-zipkin/src/main/java/io/jaegertracing/zipkin/V2SpanConverter.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2018, The Jaeger Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.jaegertracing.zipkin;
+
+import com.google.gson.Gson;
+import io.jaegertracing.Constants;
+import io.jaegertracing.LogData;
+import io.jaegertracing.Span;
+import io.jaegertracing.SpanContext;
+import io.jaegertracing.Tracer;
+import io.opentracing.tag.Tags;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Converts a Jaeger span to a Zipkin2 span.
+ */
+public class V2SpanConverter {
+
+  private static final Gson gson = new Gson();
+
+  public static zipkin2.Span convertSpan(Span span) {
+    Tracer tracer = span.getTracer();
+    zipkin2.Endpoint host = zipkin2.Endpoint.newBuilder()
+        .ip(convertIp(tracer.getIpv4()))
+        .serviceName(tracer.getServiceName())
+        .build();
+
+    zipkin2.Endpoint peerEndpoint = extractPeerEndpoint(span.getTags());
+
+    SpanContext context = span.context();
+    zipkin2.Span.Builder builder = zipkin2.Span.newBuilder()
+            .id(Long.toHexString(context.getSpanId()))
+            .traceId(Long.toHexString(context.getTraceId()))
+            .name(span.getOperationName())
+            .parentId(Long.toHexString(context.getParentId()))
+            .debug(context.isDebug())
+            .localEndpoint(host)
+            .remoteEndpoint(peerEndpoint)
+            .kind(convertKind(span.getTags().get(Tags.SPAN_KIND.getKey())))
+            .timestamp(span.getStart())
+            .duration(span.getDuration());
+
+    buildAnnotations(span, builder);
+    buildTags(span, builder);
+
+    return builder.build();
+  }
+
+  private static zipkin2.Span.Kind convertKind(Object kind) {
+    if (Tags.SPAN_KIND_SERVER.equals(kind)) {
+      return zipkin2.Span.Kind.SERVER;
+    } else if (Tags.SPAN_KIND_CLIENT.equals(kind)) {
+      return zipkin2.Span.Kind.CLIENT;
+    } else if (Tags.SPAN_KIND_CONSUMER.equals(kind)) {
+      return zipkin2.Span.Kind.CONSUMER;
+    } else if (Tags.SPAN_KIND_PRODUCER.equals(kind)) {
+      return zipkin2.Span.Kind.PRODUCER;
+    } else {
+      return null;
```

Comment:
Is this an expected case? If not, a log statement should be produced, like, "Could not identify the span kind during conversion. Skipping."

Reply:
IIRC internal spans has a null kind.

Reply:
yes they have kind set to  `null`